extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/devconsole/InvocationInterceptor.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.arc.runtime.devconsole;
+
+import javax.annotation.Priority;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.Intercepted;
+import javax.inject.Inject;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InjectableBean;
+import io.quarkus.arc.ManagedContext;
+import io.quarkus.arc.runtime.devconsole.Invocation.Builder;
+
+@Priority(Interceptor.Priority.LIBRARY_BEFORE)
+@Monitored
+@Interceptor
+public class InvocationInterceptor {
+
+    @Intercepted
+    InjectableBean<?> bean;
+
+    @Inject
+    Instance<InvocationTree> invocationTree;
+
+    @AroundInvoke
+    public Object monitor(InvocationContext context) throws Exception {
+        ManagedContext requestContext = Arc.container().requestContext();
+        if (requestContext.isActive()) {
+            InvocationTree tree = invocationTree.get();
+            return proceed(tree.invocationStarted(bean, context.getMethod()), context, tree);
+        } else {
+            return proceed(new Builder(), context, null);
+        }
+    }
+
+    Object proceed(Invocation.Builder builder, InvocationContext context, InvocationTree tree) throws Exception {
+        long start = System.nanoTime();
+        Object result = context.proceed();
+        builder.setDuration(System.nanoTime() - start);
+        if (tree != null) {
+            tree.invocationCompleted();
```

Comment:
At this point, a user could have disabled/restarted request context. You should probably re-check if it's enabled.

Reply:
Hm, that's a good point. I think that there should be also a `@PreDestroy` callback on the invocation tree to flush the data if a request context is terminated unexpectedly...

Reply:
Ok, the current version should be more robust, although not perfect. But it's probably the best we can do if someone touches the request context manually...